index.js

```diff
@@ -1 +1,120 @@
 // implement your API here
+const express = require('express');
+const db = require('./data/db');
+const server = express();
+server.use(express.json());
+
+// Delivers 'server running' response:
+server.get('/', (req, res) => {
+  res.send('Server running...');
+});
+
+// Fetches users array:
+server.get('/api/users', (req, res) => {
+  db.find()
+    .then(users => {
+      if (users) {
+        res.status(200).json(users);
+      } else {
+        res.status(404).json({ message: 'no users to return' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The users information could not be retrieved.' });
+    });
+});
+
+// Fetches a user:
+server.get('/api/users/:id', (req, res) => {
+  const id = req.params.id;
+  db.findById(id)
+    .then(user => {
+      if (user) {
+        res.status(200).json(user);
+      } else {
+        res
+          .status(404)
+          .json({ message: 'The user with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The user information could not be retrieved.' });
+    });
+});
+
+// Adds a user:
+server.post('/api/users', (req, res) => {
+  const user = req.body;
+
+  // I can't figure out the syntax to check user.name and user.bio inside the conditonal below, but defining them here works:
+  const name = req.body.name;
+  const bio = req.body.bio;
```

Comment:
good, and you do need to check to make sure these values were included in the request body, but remember you can also save some time with `const {name, bio} = req.body;`